### PR TITLE
Start of the Authentication service 

### PR DESCRIPTION
**Pull request by dbrasseur-aneo**
- Added gRPC authentication service
  - Endpoint to know your current permissions
- Added tests for this service
- Fixed badly passed along claims store

### CODE REVIEW

**Comment by dbrasseur-aneo** [comment]
Missing some doc, will add them

File: Common/src/Auth/Authentication/Authenticator.cs [hunk 1]
```diff
@@ -216,9 +217,14 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
     var keyHash = cacheKey.GetHashCode();
 
     var identity = cache_.Get(cacheKey);
-    if (identity != null)
+    if (identity is
```
Comment (aneojgurhem) [inline Common/src/Auth/Authentication/Authenticator.cs:220]: Use `is not` please !

File: Common/src/Auth/Authentication/UserIdentity.cs [hunk 2]
```diff
@@ -43,8 +43,12 @@ public class UserIdentity : ClaimsPrincipal
   /// <param name="authenticationType">Scheme by which the user is authenticated</param>
   public UserIdentity(UserAuthenticationResult userAuth,
                       string?                  authenticationType)
-    : base(new ClaimsIdentity(userAuth.Permissions.Select(perm => new Permission(perm).Claim),
-                              authenticationType))
+    : base(userAuth.Permissions.Select(perm => new Permission(perm).Claim)
```
Comment (aneojgurhem) [inline Common/src/Auth/Authentication/UserIdentity.cs:46]: It could use some explanation !

File: Common/src/Auth/Authentication/UserIdentity.cs [hunk 3]
```diff
@@ -76,4 +72,26 @@ public class UserIdentity : ClaimsIdentity
...
+    // Add the roles of the user. This is done by adding claims with type ClaimTypes.Role and with a value corresponding to the role.
+    // This claim will be used when ClaimPrincipal.IsInRole is called.
+    claims = claims.Concat(userAuth.Roles.Select(r => new Claim(ClaimTypes.Role,
+                                                                r)));
+
+    // Add the name of the user. This is done by adding claims with type ClaimTypes.Name and with a value corresponding to the username.
+    // This claim will be used when ClaimPrincipal.Name is called.
+    claims = claims.Append(new Claim(ClaimTypes.Name,
+                                     userAuth.Username));
+    return claims.ToList();
```
Comment (aneojgurhem) [inline Common/src/Auth/Authentication/UserIdentity.cs:95]: This ToList does not seem necessary as it was added in this function.

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**